substrate/frame/contracts/src/benchmarking/mod.rs

```diff
@@ -629,6 +629,27 @@ mod benchmarks {
 		}
 	}
 
+	#[benchmark(pov_mode = Measured)]
+	fn seal_caller_2() {
+		let mut setup = CallSetup::<T>::default();
+		let (mut ext, _) = setup.ext();
+		let mut runtime = crate::wasm::Runtime::new(&mut ext, vec![]);
+		let mut memory = [0u8; 36];
+		memory[0..4].copy_from_slice(&32u32.to_le_bytes());
+
+		let result;
+		#[block]
+		{
+			result = crate::wasm::BenchEnv::seal0_caller(&mut runtime, &mut memory, 4, 0);
+		}
```

Comment:
> One solution would be to make [Runtime](https://github.com/paritytech/polkadot-sdk/blob/e81322bc3e8192b536067fed3ef9e20f2752c376/substrate/frame/contracts/src/wasm/runtime.rs#L375-L380) generic over the memory access so we can mock it from the benchmarks.

@athei do we really need to make memory generic? Doesn't this approach above just work?

Reply:
ah nvm I guess we need it so that we abstract gas counting for when we read and write to memory sandbox